src/builders.ts

```diff
@@ -79,14 +73,15 @@ export class BomBuilder {
     this.metaComponentType = options.metaComponentType ?? ComponentType.Application
     this.reproducible = options.reproducible ?? false
     this.shortPURLs = options.shortPURLs ?? false
-    this.gatherLicenseTexts = options.gatherLicenseTexts
+    this.gatherLicenseTexts = options.gatherLicenseTexts ?? false
 
     this.console = console_
   }
 
   async buildFromWorkspace (workspace: Workspace): Promise<Bom> {
     // @TODO make switch to disable load from fs
     const fetchManifest: ManifestFetcher = await this.makeManifestFetcher(workspace.project)
+    const fetchLicenseEvidences: LicenseEvidenceFetcher = await this.makeLicenseEvidenceFetcher(workspace.project)
```

Comment:
I get your argument from https://github.com/CycloneDX/cyclonedx-node-yarn/pull/193#pullrequestreview-2423060733 but 2 separate fetchers means that all packages have to be extracted twice. This is a waste resource-wise.

We should therefore have only 1 fetcher but not call it manifest fetcher nor license evidence fetcher. The fetcher is something general that makes package contents available. Depending on the linker used it goes to a real file system or a virtual one like the content of ZIP files.

What do you think about letting the fetcher take multiple extractors? Then the fetcher could open the packages (using `fetcher.fetch(...)`), call all extractors and close the packages (using `releaseFs()`). In between one extractor could get the package.json while the other could search for license evidence files.